source/agora/test/Base.d

```diff
@@ -560,6 +571,68 @@ public class TestNode : Node, TestAPI
     {
         return new EnrollmentManager(":memory:", node_config.key_pair);
     }
+}
+
+/// A FullNode which also implements test routines in TestAPI
+public class TestFullNode : FullNode, TestAPI
+{
+    ///
+    mixin TestNodeMixin!();
+
+    ///
+    public this (Config config, Registry* reg)
+    {
+        assert(!config.node.is_validator);
+        this.registry = reg;
+        super(config);
+    }
+
+    /// FullNode does not implement this
+    public override void broadcastPreimage (uint height)
+    {
+        assert(0);
+    }
+
+    /// FullNode does not implement this
+    public override Enrollment createEnrollmentData ()
+    {
+        assert(0);
+    }
+
+    /// FullNode does not implement this
+    public override void updateEnrolledHeight (Hash enroll_key, ulong height)
+    {
+        assert(0);
+    }
```

Comment:
Those methods shouldn't be available if they aren't implemented then

Reply:
`public class TestFullNode : FullNode, TestAPI`
Here, `TestAPI` is inheritance interface `TestAPI: ValidatorAPI`
I think there is nothing we can do about the current TestAPI inheritance structure.
Do you have a better idea?